lib/hsa/mcwamp_hsa.cpp

```diff
@@ -1599,9 +1610,9 @@ class HSAQueue final : public KalmarQueue
         //
```

Comment:
as the algorithm becomes much more complicated with the introduction of `asyncOps_offset` field, I think it's prudent to have a paragraph of comment to explain the algorithm better.